src/with-theme.js

```diff
@@ -3,37 +3,52 @@ import React, {Component} from 'react'
 import {CHANNEL} from './constants'
```

Comment:
The changes in this file are mostly to refactor it to be more in line with what we had for regular GlamorousComponents. I believe that we always intended this to work the same way that the GlamorousComponent implementation worked and it didn't in some subtle ways. Good to bring these into alignment.